sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -73,8 +73,10 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    * @since 1.4.0
    */
   def schema(schema: StructType): DataFrameReader = {
-    val replaced = CharVarcharUtils.failIfHasCharVarchar(schema).asInstanceOf[StructType]
-    this.userSpecifiedSchema = Option(replaced)
+    if (schema != null) {
```

Comment:
Seems fine. It'd be OK IMHO to require it to be non-null too.

Reply:
I'd also prefer not allowing null, but no strong voice.